src/Solocraft.cpp

```diff
@@ -77,106 +490,722 @@ class solocraft_player_instance_handler : public PlayerScript {
+                difficulty = D543;
+			}
+            else if (map->GetId() == 545) {
+                difficulty = D545;
+			}
+            else if (map->GetId() == 546) {
+                difficulty = D546;
+			}
+            else if (map->GetId() == 547) {
+                difficulty = D547;
+			}
+            else if (map->GetId() == 552) {
+                difficulty = D552;
+			}
+            else if (map->GetId() == 553) {
+                difficulty = D553;
+			}
+            else if (map->GetId() == 554) {
+                difficulty = D554;
+			}
+            else if (map->GetId() == 555) {
+                difficulty = D555;
+			}
+            else if (map->GetId() == 556) {
+                difficulty = D556;
+			}
+            else if (map->GetId() == 557) {
+                difficulty = D557;
+			}
+            else if (map->GetId() == 558) {
+                difficulty = D558;
+			}
+            else if (map->GetId() == 560) {
+                difficulty = D560;
+			}
+            else if (map->GetId() == 568) {
+                difficulty = D568;
+			}
+            else if (map->GetId() == 585) {
+                difficulty = D585;
+			}			
+			//WOTLK 5 Man
+            else if (map->GetId() == 574) {
+                difficulty = D574;
+			}
+            else if (map->GetId() == 575) {
+                difficulty = D575;
+			}
+            else if (map->GetId() == 576) {
+                difficulty = D576;
+			}
+            else if (map->GetId() == 578) {
+                difficulty = D578;
+			}
+            else if (map->GetId() == 595) {
+                difficulty = D595;
+			}
+            else if (map->GetId() == 599) {
+                difficulty = D599;
+			}
+            else if (map->GetId() == 600) {
+                difficulty = D600;
+			}
+            else if (map->GetId() == 601) {
+                difficulty = D601;
+			}
+            else if (map->GetId() == 602) {
+                difficulty = D602;
+			}
+            else if (map->GetId() == 604) {
+                difficulty = D604;
+			}
+            else if (map->GetId() == 608) {
+                difficulty = D608;
+			}
+            else if (map->GetId() == 619) {
+                difficulty = D619;
+			}
+            else if (map->GetId() == 632) {
+                difficulty = D632;
+			}
+            else if (map->GetId() == 650) {
+                difficulty = D650;
+			}
+            else if (map->GetId() == 658) {
+                difficulty = D658;
+			}
+            else if (map->GetId() == 668) {
+                difficulty = D668;
+			}			
+			//Catch Alls  ----------------------5 Dungeons and 40 Raids
             else if (map->IsDungeon()) {
                 difficulty = D5;
             }
+            else if (map->IsRaid()) {
+                difficulty = D40;
+            }			
         }
-        return difficulty;
+        return difficulty; //return the specific dungeon's difficulty offset
     }
```

Comment:
even here, a hashmap is a lot better, you can even improve the configs variables